lib/Dialect/RTL/Ops.cpp

```diff
@@ -427,7 +427,16 @@ void AndOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
         return success();
       }
 
-      /// TODO: and(x, and(...)) -> and(x, ...) -- flatten
+      // and(x, and(...)) -> and(x, ...) -- flatten
+      if (auto andOp = inputs.back().getDefiningOp<rtl::AndOp>()) {
```

Comment:
This should check each of the `inputs` to see if any of them is an `and`, not just the last one.

I think this should also check "hasOneUse" on the `andOp`.   Not doing so can lead to IR blowups in things like:

%t1 = and(a, b, .....)
%t2 = and(t1, c)
%t3 = and(t1, d)

by duplicating the first and into the other operations.

Reply:
PTAL at the new commit. I've written test cases for when the andOp is in the beginning and middle as well, and they all pass. However, I'm not sure if there's a more "elegant" approach to this. I know there's `drop_front` and `drop_back`, but I couldn't find anything similar to `drop_nth_element`.